app/views/podcast/single-podcast/EpisodeComponent.vue

```diff
@@ -0,0 +1,208 @@
+<template>
+  <div class="episode">
+    <div class="container">
+      <div class="row episode__frame">
+        <div class="col-md-offset-3 col-md-7">
+          <iframe :src="transistorUrl"
+                  width='100%' height='180' frameborder='0' scrolling='no'
+                  seamless='true' style='width:100%; height:180px;'>
+          </iframe>
+        </div>
+      </div>
+      <div class="row">
+        <div class="col-md-offset-1 col-md-2 episode__air">
+          <div class="episode__air-date">
+            <div class="episode__air-date-text">
+              {{ $t('podcast.air_date') }}
+            </div>
+            <div class="episode__air-date-date">
+              {{ getUploadDate(podcast.uploadDate) }}
+            </div>
+          </div>
+          <div class="episode__subscribe" @click="onSubscribeClick">
+            {{ $t('podcast.subscribe') }}+
+          </div>
+        </div>
+        <div class="col-md-5 episode__info">
+          <div class="episode__info-description" v-html="formatDescription"></div>
+        </div>
+        <div class="col-md-4 episode__function">
+          <div class="episode__btn-info episode__function-download" @click="() => onDownloadClick(podcast)">
+            <img src="/images/pages/podcast/IconDownload.svg" alt="Download Icon" class="episode__download-icon episode__icon">
+            <span class="episode__listen episode__btn-text episode__btn-text-hor">{{ $t('podcast.download') }}</span>
+          </div>
+          <div class="episode__btn-info" @click="() => onTranscriptClick(podcast)">
+            <img src="/images/pages/podcast/IconTranscript.svg" alt="Transcript Icon" class="episode__transcript-icon episode__icon">
+            <span class="episode__listen episode__btn-text episode__btn-text-hor">{{ $t('podcast.transcript') }}</span>
+          </div>
+        </div>
+      </div>
+    </div>
+    <audio-player-component
+      :transistor-episode-id="podcast.transistorEpisodeId"
+      v-if="showPlayModal"
+      @close="showPlayModal = false"
+    />
+    <subscribe-modal
+      v-if="showSubscribeModal"
+      @close="showSubscribeModal = false"
+    />
+  </div>
+</template>
+
+<script>
+import AudioPlayerComponent from '../AudioPlayerComponent'
+import SubscribeModal from '../SubscribeModal'
+import { fullFileUrl } from '../podcastHelper'
+import uploadDateMixin from '../uploadDateMixin'
+const marked = require('marked')
+export default {
+  name: 'EpisodeComponent',
+  props: {
+    podcast: {
+      type: Object,
+      required: true
+    }
+  },
+  components: {
+    AudioPlayerComponent,
+    SubscribeModal
+  },
+  data () {
+    return {
+      showPlayModal: false,
+      showSubscribeModal: false
+    }
+  },
+  mixins: [ uploadDateMixin ],
+  methods: {
+    onListenClick (podcast) {
+      console.log('listen', podcast, this.showPlayModal)
```

Comment:
I would love to have `trackEvent` or even `trackActivity` calls on these actions.

Reply:
This is old modal code, I will remove it.

On the topic of tracking, transistor shows tracking on its side where plays from the embed-player will be tagged. Though, we can't differentiate on whether that embed player was on listing page or single podcast page.